### PR TITLE
Synonym

### PR DESCRIPTION
**Pull request by Alnusjaponica**
## Motivation
Implement `langcheck.augment.synonym()`

### CODE REVIEW

**Comment by Alnusjaponica** [comment]
I opened this PR as #25 is merged. PTAL.

File: tests/augment/test_synonym.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        (["Hello, world!"], ["Hullo, world!"]),
+        (["Hello, world!", "I have a pen. I have an apple"
+         ], ["Hullo, world!", "I have a pen. I have an malus pumila"]),
+    ],
+)
+def test_synonym(texts: list[str], expected: list[str]):
+    seed = 42
+    random.seed(seed)
+    actual = synonym(texts)
+    print(actual)
```
Comment (liwii) [inline tests/augment/test_synonym.py:22]:
```suggestion
```

File: src/langcheck/augment/en/_synonym.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        is acceptable. Some of the more useful ones from `nlpaug` document are
+        listed below:
+
+          - ``aug_p`` (float): Percentage of word will be augmented.
+          - ``aug_min``: Minimum number of word will be augmented.
+          - ``aug_max``: Maximum number of word will be augmented.
+
+        Note that the default values for these arguments are different from the
+        ``nlpaug`` defaults. To be more specific, the default values for
+        ``aug_p`` to be `0.1` and ``aug_char_max`` to be `None`. See the source
```
Comment (liwii) [inline src/langcheck/augment/en/_synonym.py:31]:
```suggestion
        ``aug_p`` to be `0.1` and ``aug_max`` to be `None`. See the source
```

**Review by liwii — APPROVED** [review]
LGTM after resolving two minor comments!!

**Pull request merged**